custom_components/sat/coordinator.py

```
@@ -265,21 +354,48 @@
+            current_time = monotonic()
+            if not self._boiler_temperatures or self._boiler_temperatures[-1][0] != current_time:
+                self._boiler_temperatures.append((current_time, self.boiler_temperature))
 
-        # Clear up any values that are older than the specified age
-        while self.boiler_temperatures and current_time - self.boiler_temperatures[0][0] > timedelta(seconds=MAX_BOILER_TEMPERATURE_AGE):
-            self.boiler_temperatures.pop()
+        # Remove old temperature records beyond the allowed age
+        self._boiler_temperatures = [
+            (timestamp, temperature)
+            for timestamp, temperature in self._boiler_temperatures
+            if seconds_since(timestamp) <= MAX_BOILER_TEMPERATURE_AGE
```

Ruff (0.8.2)

`MAX_BOILER_TEMPERATURE_AGE` may be undefined, or defined from star imports

(F405)